cuda_core/cuda/core/experimental/_device.py

```diff
@@ -1160,6 +1160,16 @@ def __int__(self):
     def __repr__(self):
         return f"<Device {self._id} ({self.name})>"
 
+    def __reduce__(self):
+        return Device._reconstruct, (self.device_id,)
+
+    @staticmethod
+    def _reconstruct(device_id):
+        device = Device(device_id)
+        if not device._has_inited:
+            device.set_current()
```

Comment:
I would prefer that we keep this explicit for now and avoid diverging behaviors between parent and children processes. Is it possible?
```suggestion
```

Reply:
> The resources are deserialized before entering `child_main`

This is horrifying...